libpromises/pipes_unix.c

```diff
@@ -142,6 +169,113 @@ static pid_t CreatePipeAndFork(const char *type, int *pd)
 
 /*****************************************************************************/
 
+static pid_t CreatePipeAndFork(const char *type, int *pd)
+{
+    IOPipe pipes[2] = {{0}};
+    pipes[0].type = type;
+    
+    pid_t pid = GenericCreatePipeAndFork(pipes);
+    
+    pd[0] = pipes[0].pipe_desc[0];
+    pd[1] = pipes[0].pipe_desc[1];
+
+    return pid;
+}
+
+/*****************************************************************************/
+
+static pid_t CreatePipesAndFork(const char *type, int *pd, int *pdb)
+{
+    IOPipe pipes[2] = {{0}};
```

Comment:
0 is a bad choice for initialiser of descriptors, as it's a valid file descriptor (albeit usually in use as stdin); -1 is a better choice for the pipe_desk[] fields' default (and as the sentinel to compare against for failure).